geo/src/algorithm/offset/offset_trait.rs

```diff
@@ -0,0 +1,336 @@
+use super::line_intersection::FalseIntersectionPointType::{
+    BeforeStart,
+    AfterEnd,
+};
+use super::line_intersection::LineSegmentIntersectionType::{
+    FalseIntersectionPoint, TrueIntersectionPoint,
+};
+use super::line_intersection::{
+    line_segment_intersection_with_relationships, LineIntersectionWithParameterResult,
+};
+use super::slice_itertools::pairwise;
+
+use crate::{
+    Coord,
+    CoordFloat,
+    // Kernel,
+    // Orientation,
+    Line,
+    LineString,
+    MultiLineString,
+    // Polygon,
+};
+
+/// # Offset Trait
+///
+/// Signed offset of Geometry assuming cartesian coordinate system.
+///
+/// This is a cheap offset algorithm that is suitable for flat coordinate systems
+/// (or if your lat/lon data is near the equator)
+///
+/// My Priority for implementing the trait is as follows:
+/// - [X] Line<impl CoordFloat>
+/// - [X] LineString<impl CoordFloat>
+/// - [X] MultiLineString<impl CoordFloat>
+/// - ... maybe some closed shapes like triangle, polygon?
+///
+/// The following are a list of known limitations,
+/// some may be removed during development,
+/// others are very hard to fix.
+///
+/// - No checking for zero length input.

```

Comment:
Is there a specific difficulty you've encountered while trying to implement a check against this?